dotnet-gitchanges/Program.cs

```diff
@@ -117,8 +117,11 @@ static void Main(string[] args)
                         var overrideFileReader = new FileReader<OverrideChange>(appConfig.Repository.OverrideSource, new OverrideSourceRowParser(Console.Error));
                         idToOverrideChange = overrideFileReader.Values().ToDictionary<OverrideChange, string, IChange>(change => change.Id, change => change);
                     }
-                    
-                    var gitReader = new GitReader<IChange>(repo, new DefaultCommitParser(appConfig.Parsing), idToOverrideChange);
+
+                    var commitParser = appConfig.UseGitTags ?
+                        (ICommitParser<IChange>)new DefaultCommitParser(appConfig.Parsing) :
+                        (ICommitParser<IChange>)new GitTagCommitParser(appConfig.Parsing, repo);

```

Comment:
Instead of passing the repo into the parser, why not just pass a map of Tag Sha -> Tag in. Then tests should be easy.

Reply:
I can make that - should be fine. I'll also try and integrate the two into just `DefaultCommitParser` - still with the bool for it to actually look at the git tags though.